src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -5619,27 +5611,20 @@ public static void Main()
             var compVerifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
             compVerifier.VerifyIL("C.M1", """
 {
-  // Code size       40 (0x28)
+  // Code size       33 (0x21)
   .maxstack  1
-  .locals init (bool V_0)
   IL_0000:  ldarg.0
   IL_0001:  isinst     "int"
   IL_0006:  brtrue.s   IL_0012
   IL_0008:  ldarg.0
   IL_0009:  isinst     "long"
   IL_000e:  brtrue.s   IL_0012
-  IL_0010:  br.s       IL_0016
-  IL_0012:  ldc.i4.1
-  IL_0013:  stloc.0
-  IL_0014:  br.s       IL_0018
-  IL_0016:  ldc.i4.0
-  IL_0017:  stloc.0
-  IL_0018:  ldloc.0
-  IL_0019:  brtrue.s   IL_0022
-  IL_001b:  ldstr      "False"
-  IL_0020:  br.s       IL_0027
-  IL_0022:  ldstr      "True"
-  IL_0027:  ret
+  IL_0010:  br.s       IL_0014
+  IL_0012:  br.s       IL_001b
```

Comment:
>IL_0012:  br.s       IL_001b [](http://example.com/codeflow?start=2&length=28)

I am curious if this intermediate branch will be optimized away in Release build

Reply:
It is. The release output is asserted a few lines below.